stumpy/core.py

```diff
@@ -107,7 +109,7 @@ def check_dtype(a, dtype=np.floating):  # pragma: no cover
     return True
 
 
-def transpose_dataframe(a):  # pragma: no cover
+def transpose_dataframe(a: np.ndarray) -> np.ndarray:  # pragma: no cover
```

Comment:
This type annotation seems to not capture the full range of inputs, It seems like it should be more like

```python
def transpose_dataframe(a: Union[np.ndarray, "DataFrame", "Series"]) -> Union[np.ndarray, "DataFrame", "Series"]:
```

I don't know if that will cause mypy problems though

Reply:
I guess we really only want a Pandas DataFrame (not Series or ndarray) as input to this function so the docstring needs to be changed to a Pandas DataFrame.

Reply:
All other inputs except for DataFrame is essentially a no-op

Reply:
Hmm, perhaps this should be `Any` type

Reply:
@cydave Feel free to chime in. 

@blester Are there other type hints that aren't correct or could be improved? I'm a total novice when it comes to this so I'd appreciate your input too!

Reply:
Because I tried strictly following the docs that were present and don't know the code base too well I'm not aware of all the inputs. My advice would be to check all references to this function and then decide upon a type that is acceptable for the invocation of this function.

I can take a look and try and see if @blester125's suggestion would work without complains from mypy.

Reply:
Hmmm, I'm struggling a bit here. We certainly do not want to import Pandas as it should not and is not dependency of STUMPY. I think I'll just merge this PR as is and deal with improving this later. Thank you for your input!

Reply:
Right, I knew pandas wasn't a dependency which is why i was suggesting to use the string names instead of the actual objects (i still don't know if that plays nicely with mypy but it is how you type a method on a class that takes a member of that class so it seems like it is supported) that way STUMPY wouldn't have a dep on pandas but someone who is typechecking their code that passes in a data frame wouldn't get errors.

Reply:
mypy must be able to resolve the type you define it. When you wrap it in quotes it still tries to resolve the type after having imported everything. This means mypy would complain:

```
$ mypy stumpy       
stumpy/core.py:112: error: Name 'DataFrame' is not defined
stumpy/core.py:112: error: Name 'Series' is not defined
Found 2 errors in 1 file (checked 11 source files)
```

An ugly workaround would be to wrap the pandas import like so:
```
if False:
    import pandas as pd
```

This way you can use the type annotation but do not depend on pandas directly (as it will never be imported).